### PR TITLE
Rule S2953: Add C#9 partial method support

### PR DESCRIPTION
**Pull request by dmitry-golovinov-sonarsource**

### CODE REVIEW

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/DisposeNotImplementingDisposeTest.cs [hunk 1]
```diff
@@ -34,7 +34,8 @@ public class DisposeNotImplementingDisposeTest
 #if NET
         [TestMethod]
         public void DisposeNotImplementingDispose_CSharp9() =>
-            Verifier.VerifyAnalyzerFromCSharp9Console(@"TestCases\DisposeNotImplementingDispose.CSharp9.cs", new DisposeNotImplementingDispose());
+            Verifier.VerifyAnalyzerFromCSharp9Console(new[] { @"TestCases\DisposeNotImplementingDispose.CSharp9.Part1.cs", @"TestCases\DisposeNotImplementingDispose.CSharp9.Part2.cs", },
+                new DisposeNotImplementingDispose());
```
[the remark this answers is not in the document]
Comment (costin-zaharia-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/DisposeNotImplementingDisposeTest.cs:38]: As far as I remember we discussed this already at how we work and decided to keep the alignment. I would not change the rule unless we all agree at the next meeting

[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/DisposeNotImplementingDisposeTest.cs:38]: I remember discussing method declarations (2020-06-24), not invocations. That one is currently not enforced, as we agreed that we need tooling for that and we don't have it.

Doing this on the invocation side wasn't discussed as far as I remember. And it has significant maintenance cost as we saw recently on another PR with batch renaming.

[the remark this answers is not in the document]
Comment (costin-zaharia-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/DisposeNotImplementingDisposeTest.cs:38]: Ah, I though we will handle them together in the same way. Regarding the tooling Rider supports this nicely. I pretty sure we can find a setting or a plugin to do this also in VS.

[the remark this answers is not in the document]
Comment (dmitry-golovinov-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/DisposeNotImplementingDisposeTest.cs:38]: Let's leave it as it is and I will add this point to how we work. Ok?

[the remark this answers is not in the document]
Comment (dmitry-golovinov-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/DisposeNotImplementingDisposeTest.cs:38]: Added it to next how we work

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**